src/workerd/api/worker-rpc.c++

```diff
@@ -1429,15 +1426,24 @@ public:
 
   TargetInfo getTargetInfo(Worker::Lock& lock, IoContext& ioCtx) override {
     return {
-      .target = object.getHandle(lock),
+      .target = handles->object.getHandle(lock),
       .envCtx = kj::none,
       .allowInstanceProperties = allowInstanceProperties,
     };
   }
 
 private:
-  jsg::JsRef<jsg::JsObject> object;
-  kj::Maybe<jsg::V8Ref<v8::Function>> dispose;
+  struct Handles {
+    jsg::JsRef<jsg::JsObject> object;
+    kj::Maybe<jsg::V8Ref<v8::Function>> dispose;
+
+    Handles(jsg::Lock& js, jsg::JsObject object, kj::Maybe<v8::Local<v8::Function>> dispose)
+        : object(js, object),
+          dispose(dispose.map([&](v8::Local<v8::Function> func) {
+            return jsg::V8Ref<v8::Function>(js.v8Isolate, func);
+          })) {}
+  };
+  ReverseIoOwn<Handles> handles;
```

Comment:
A comment here explaining why the `ReverseIoOwn<T>` is necessary here would be helpful for those coming into this code in the future.